shared-bindings/_bleio/Adapter.c

```diff
@@ -182,11 +189,13 @@ STATIC mp_obj_t bleio_adapter_start_advertising(mp_uint_t n_args, const mp_obj_t
     }
 
     bool connectable = args[ARG_connectable].u_bool;
+    bool anonymous = args[ARG_anonymous].u_bool;
+    uint32_t timeout = args[ARG_timeout].u_int;
```

Comment:
Since the common_hal routine will ignore `anonymous` if there's no timeout, could you add a validation check for this, and complain if `anonymous` is set without a timeout?

Reply:
Okay, if you specify anonymous, it now requires you specify a timeout, rather than picking one.

Reply:
Sorry, I misunderstood the `common_hal` routine. I thought it was ignoring the `anonymous` if there as no timeout, but that was just because I was missing some code due to the way the diff displayed. It's fine to not specify the timeout if `anonymous` is `True`, as long as the constructor's documentation says that, and says what the timeout will be. So I believe your old logic is OK, if it's as stated above; just add more doc to the `timeout` arg description.